src/Debugger/SourceContext.php

```diff
@@ -0,0 +1,26 @@
+<?php
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Debugger;
+
+/**
+ * A SourceContext is a reference to a tree of files. A SourceContext together
+ * with a path point to a unique revision of a single file or directory.
+ */
+abstract class SourceContext
```

Comment:
I'm not sure I like an empty abstract class here. What are your thoughts on an interface with a `getContext` (or similar) method? If we like, `jsonSerialize` can just call `getContext`. This will also be nice for when we support other transports as we won't need be serializing to json for gRPC.